### PR TITLE
feat: a simple service exposing Search API

### PR DESCRIPTION
**Pull request by jachro**
This PR brings some initial implementation of the Search API service.

### CODE REVIEW

File: modules/http4s-avro/src/main/scala/io/renku/search/http/avro/AvroEntityCodec.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      )
+
+  def encodeEntity[F[_], A: AvroJsonEncoder]: EntityEncoder[F, A] =
+    EntityEncoder.simple(`Content-Type`(application.json))(a =>
+      Chunk.byteVector(AvroJsonEncoder[A].encode(a))
+    )
+
+  trait Implicits:
+
+    implicit def entityDecoder[F[_]: Async, A: AvroJsonDecoder]: EntityDecoder[F, A] =
```
Comment (eikek) [inline modules/http4s-avro/src/main/scala/io/renku/search/http/avro/AvroEntityCodec.scala:62]: We can now use `given`:) (I might forgot to change that myself when I used code from somewhere else:))

File: modules/search-api/src/main/scala/io/renku/search/api/SearchApiImpl.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import io.renku.avro.codec.all.given
+import io.renku.avro.codec.json.AvroJsonEncoder
+import io.renku.search.http.avro.AvroEntityCodec.Implicits.entityEncoder
+import io.renku.search.solr.client.SearchSolrClient
+import io.renku.search.solr.documents.Project as SolrProject
+import org.http4s.Response
+import org.http4s.dsl.Http4sDsl
+import scribe.Scribe
+
+private class SearchApiImpl[F[_]: Async: Scribe](solrClient: SearchSolrClient[F])
```
Comment (eikek) [inline modules/search-api/src/main/scala/io/renku/search/api/SearchApiImpl.scala:33]: I know we have it like this in graph passing in the logger from outside. I'm not sure why that is good? Why wouldn't we want to have different loggers per class?

File: modules/search-api/src/main/scala/io/renku/search/api/SearchApiImpl.scala [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      .findProjects(phrase)
+      .map(toApiModel)
+      .map(toAvroResponse)
+      .handleErrorWith(errorResponse(phrase))
+
+  private given AvroJsonEncoder[List[ApiProject]] =
+    AvroJsonEncoder.encodeList[ApiProject](ApiProject.SCHEMA$)
+
+  private def toAvroResponse(entities: List[ApiProject]): Response[F] =
+    Response[F](Ok)
```
Comment (eikek) [inline modules/search-api/src/main/scala/io/renku/search/api/SearchApiImpl.scala:48]: with `Http4sDsl` in scope, I think writing `Ok(entities)` should work as well. A separate method might then not be needed?

**Review by eikek — APPROVED** [review]

File: modules/http4s-avro/src/main/scala/io/renku/search/http/avro/AvroEntityCodec.scala (hunk 1 above)
Comment (jachro) [reply modules/http4s-avro/src/main/scala/io/renku/search/http/avro/AvroEntityCodec.scala:62]: Yes, I used metrics collector as a template ;)

**Pull request merged**